crates/simulations/src/cozy/agents/altruistic_supplier.rs

```diff
@@ -0,0 +1,63 @@
+use std::{borrow::Cow, sync::Arc};
+
+use simulate::{address::Address, agent::Agent, state::State, u256::U256};
+
+use crate::cozy::{
+    runner::{ProtocolContracts, SetContracts},
+    world::{CozyUpdate, CozyWorld},
+};
+
+pub struct AltruisticSupplier {
+    _name: Cow<'static, str>,
+    address: Address,
+    protocol: Arc<ProtocolContracts>,
+    set: Arc<SetContracts>,
+}
+
+impl AltruisticSupplier {
+    pub fn new(
+        _name: Cow<'static, str>,
+        address: Address,
+        protocol: Arc<ProtocolContracts>,
+        set: Arc<SetContracts>,
+    ) -> Self {
+        Self {
+            _name,
+            address,
+            protocol,
+            set,
+        }
+    }
+}
+
+impl Agent<CozyUpdate, CozyWorld> for AltruisticSupplier {
```

Comment:
What's the reasoning for only having this agent supplying during the activation step?

Reply:
Altruistic supplier is a special case supplier that always supplies assets to the Set at the beginning of the sim to bootstrap the markets and unlike the standard supplier cannot withdraw assets. Basically, they exist to make sure the set always has some assets available. So, we only need them to transact during the activation step.